__tests__/app.test.js

```diff
@@ -276,53 +303,97 @@ describe("app.js", () => {
+          expect(response.body.commentInserted.comment_id).toBe(19);
+          expect(response.body.commentInserted).toHaveProperty("body");
+          expect(response.body.commentInserted.body).toBe(
+            "A-ha! Guess who's back in the big-time?"
+          );
+          expect(response.body.commentInserted).toHaveProperty("article_id");
+          expect(response.body.commentInserted.article_id).toBe(3);
+          expect(response.body.commentInserted).toHaveProperty("author");
+          expect(response.body.commentInserted.author).toBe("icellusedkars");
+          expect(response.body.commentInserted).toHaveProperty("votes");
+          expect(response.body.commentInserted.votes).toBe(0);
+          expect(response.body.commentInserted).toHaveProperty("created_at");
+        });
+    });
+
+    test("Status 404: returns an error if post request made to a non-existent article_id", () => {
+      return request(app)
+        .post("/api/articles/999999/comments")
+        .send({
+          username: "A. Partridge",
+          body: "A-ha! Guess who's back in the big-time?",
+        })
         .expect(404)
         .then((response) => {
           expect(response.body.msg).toBe("article_id not found");
         });
     });
 
-    test("Status 404 - returns a message with 'specified article_id has no comments' for article_id 2", () => {
+    test("Status 400: returns an error if post request made to an article_id which is not a number", () => {
       return request(app)
-        .get("/api/articles/2/comments")
-        .expect(404)
+        .post("/api/articles/banana/comments")
+        .send({
+          username: "A. Partridge",
+          body: "A-ha! Guess who's back in the big-time?",
+        })
+        .expect(400)
         .then((response) => {
-          expect(response.body.msg).toBe(
-            "specified article_id has no comments"
-          );
+          expect(response.body.msg).toBe("article_id is not a number");
         });
     });
 
-    test("Status 404: returns a message 'article_id not found' if article_id doesn't exist", () => {
+    test("Status 400: returns an error if post request is missing username", () => {
       return request(app)
-        .get("/api/articles/banana/comments")
-        .expect(404)
+        .post("/api/articles/1/comments")
+        .send({ body: "A-ha! Guess who's back in the bigtime?" })
+        .expect(400)
         .then((response) => {
-          expect(response.body.msg).toBe("article_id not found");
+          expect(response.body.msg).toBe("comment is missing username");
+        });
+    });
+
+    test("Status 400: returns an error if post request is missing body", () => {
+      return request(app)
+        .post("/api/articles/1/comments")
+        .send({ username: "A. Partridge" })
+        .expect(400)
+        .then((response) => {
+          expect(response.body.msg).toBe("comment is missing body");
         });
     });
   });
```

Comment:
Would like to see a test where we send both a username and body to a valid article, but the username does not exist in the database please :)